x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/Resampler.java

```diff
@@ -50,7 +50,7 @@ public int adjustSampleCount(int originalCount) {
         }
         // Adjust the sample counts from down-sampled to fully sampled.
         // Be aware that downsampling drops entries from stackTraceEvents, so that
-        // the sum of the upscaled count values is less that totalCount.
-        return (int) Math.floor(rawCount / (p * adjustedSampleRate));
+        // the sum of the upscaled count values is less than totalCount.
+        return (int) Math.round(rawCount / (p * adjustedSampleRate));
```

Comment:
Can you add a comment here as to why we're rounding up?

Reply:
It's not rounding up, it just rounding. The previous rounding down isn't correct, e.g. why should we return `2` if the calculated value is `2.99` (nearly 3)? So `floor()` creates a statistical bias towards lower count values.